packages/core/src/interpreter.ts

```diff
@@ -282,13 +311,22 @@ export class Actor<
     }
     this.status = ActorStatus.Running;
 
+    const initEvent = createInitEvent(this.options.input);
+
+    this.system._sendInspectionEvent({
+      type: '@xstate.event',
+      event: initEvent,
+      sourceId: this._parent?.sessionId,
```

Comment:
I wonder if this is truly OK, is the init event received from the parent or is it raised internally? I can see arguments going both ways :p

Reply:
Conceptually, parent creates a child with input that is inherited from the parent